Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabPublicRequestHandler.h

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/Memory/SystemAllocator.h>
+#include <AzCore/RTTI/RTTI.h>
+
+#include <AzToolsFramework/Prefab/PrefabPublicInterface.h>
+#include <AzToolsFramework/Prefab/PrefabPublicRequestBus.h>
+
+namespace AzToolsFramework
+{
+    namespace Prefab
+    {
+        
+        class PrefabPublicRequestHandler final
+            : public PrefabPublicRequestBus::Handler
+        {
+        public:
+            AZ_CLASS_ALLOCATOR(PrefabPublicRequestHandler, AZ::SystemAllocator, 0);
+            AZ_RTTI(PrefabPublicRequestHandler, "{83FBDDF9-10BE-4373-B1DC-44B47EE4805C}");
+
+            static void Reflect(AZ::ReflectContext* context)
+            {
+                AZ::BehaviorContext* behaviorContext = azrtti_cast<AZ::BehaviorContext*>(context);
+                if (behaviorContext)
+                {
+                    behaviorContext->EBus<PrefabPublicRequestBus>("PrefabPublicRequestBus")
+                        ->Attribute(AZ::Script::Attributes::Scope, AZ::Script::Attributes::ScopeFlags::Automation)
+                        ->Attribute(AZ::Script::Attributes::Category, "Prefab")
+                        ->Attribute(AZ::Script::Attributes::Module, "prefab")
+                        ->Event("CreatePrefabInDisk", &PrefabPublicRequests::CreatePrefabInDisk)
+                        ->Event("CreatePrefabInMemory", &PrefabPublicRequests::CreatePrefabInMemory)
+                        ->Event("InstantiatePrefab", &PrefabPublicRequests::InstantiatePrefab)
+                        ;
+                }
+            }
+
+            void Connect()
+            {
+                m_prefabPublicInterface = AZ::Interface<Prefab::PrefabPublicInterface>::Get();
```

Comment:
Any reason why implementation is not in a separate cpp file like we generally do? Unless this is required for tests to work for some reason, I suggest having .h and .cpp files for this class.

Reply:
Fixed.